azurerm/resource_arm_automation_module.go

```diff
@@ -0,0 +1,186 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/automation/mgmt/2015-10-31/automation"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmAutomationModule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAutomationModuleCreateUpdate,
+		Read:   resourceArmAutomationModuleRead,
+		Update: resourceArmAutomationModuleCreateUpdate,
+		Delete: resourceArmAutomationModuleDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"automation_account_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"module_link": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
```

Comment:
>MaxItems: 1, [](start = 4, length = 12)

*suggestion* Since this object will occur at most once and the whole resource represents a module, will it be better to remove `module_link` object and just move `uri` and `hash` to the top level named as `module_uri` and `module_hash`?

Reply:
I used the same implementation as with the runbook resource.